packages/cadl-rlc-emitter/test/unit/generate/clientFactory.spec.ts

```diff
@@ -0,0 +1,145 @@
+      @server(
+        "localhost",
+        "Language Service"
+      )
+      @serviceTitle("PetStoreClient")
+      namespace PetStore;
+      `);
+      assert.ok(models);
+      assertEqualContent(
+        models!.content,
+        `
+        import { getClient, ClientOptions } from "@azure-rest/core-client";
+        import { testClient } from "./clientDefinitions";
+        
+        export default function createClient(options: ClientOptions = {}): testClient {
+        const baseUrl = options.baseUrl ?? \`localhost\`;
+        
+        const userAgentInfo = \`azsdk-js--rest/1.0.0-beta.1\`;
+        const userAgentPrefix =
+            options.userAgentOptions && options.userAgentOptions.userAgentPrefix
+            ? \`\${options.userAgentOptions.userAgentPrefix} \${userAgentInfo}\`
+            : \`\${userAgentInfo}\`;
+        options = {
+            ...options,
+            userAgentOptions: {
+            userAgentPrefix,
+            },
+        };
+        
+        const client = getClient(baseUrl, options) as testClient;
+        
+        return client;
+    }
+    `);
+    });
+    it("should handle one parameter", async () => {
+      const models = await emitClientFactoryFromCadl(`
+          @server(
+            "{Endpoint}/language",
+            "Language Service",
+            {
+              Endpoint: Endpoint,
+            }
+          )
+          @serviceTitle("PetStoreClient")
+          namespace PetStore;
+          @doc("The endpoint to use.")
+          model Endpoint is string;
+          `);
+      assert.ok(models);
+      assertEqualContent(
+        models!.content,
+        `
+          import { getClient, ClientOptions } from "@azure-rest/core-client";
+          import { testClient } from "./clientDefinitions";
+          
+          export default function createClient(
+            Endpoint: string,
+            options: ClientOptions = {}
+          ): testClient {
+            const baseUrl = options.baseUrl ?? \`\${Endpoint}/language\`;
+          
+            const userAgentInfo = \`azsdk-js--rest/1.0.0-beta.1\`;
+            const userAgentPrefix =
+              options.userAgentOptions && options.userAgentOptions.userAgentPrefix
+                ? \`\${options.userAgentOptions.userAgentPrefix} \${userAgentInfo}\`
+                : \`\${userAgentInfo}\`;
+            options = {
+              ...options,
+              userAgentOptions: {
+                userAgentPrefix,
+              },
+            };
+          
+            const client = getClient(baseUrl, options) as testClient;
+          
+            return client;
+        }
+        `
+      );
+    });
+
+    it("should handle two parameters", async () => {
+        const models = await emitClientFactoryFromCadl(`
+            @server(
+              "{Endpoint}/language/{Version}",
+              "Language Service",
+              {
+                Endpoint: Endpoint,
```

Comment:
Do you know if we could set the parameter as optional?

Reply:
Not sure I understand the question. The path parameters that @qiaozha is working on in this PR are different in that these are part of the "host", not the operation resource. These are shared across all requests, however, the operation path parameters are specific to each operation. Hope this answers your question

Reply:
Yes, thanks for your explanations!